types/firestore.d.ts

```diff
@@ -411,6 +410,216 @@ declare namespace FirebaseFirestore {
            precondition?: Precondition): Transaction;
   }
 
+  /**
+ * A Firestore BulkWriter than can be used to perform large amounts of writes in
+ * parallel. Writes to the same document will be executed sequentially.
+ *
+ * @class
+ */
+export class BulkWriter {
+  private constructor();
+
+  /**
+   * Create a document with the provided data. This single operation will fail
+   * if a document exists at its location.
+   *
+   * @param {DocumentReference} documentRef A reference to the document to be
+   * created.
+   * @param {T} data The object to serialize as the document.
+   * @returns {Promise<WriteResult>} A promise that resolves with the result of
+   * the write. Throws an error if the write fails.
+   *
+   * @example
```

Comment:
We don't really have examples in this file.

Reply:
Note that we did put the converter example in here because it is only valid in TypeScript and does not show up in JSDoc.